tests/examples/test_check_shapes_examples.py

```diff
@@ -0,0 +1,739 @@
+
+    @check_shapes(
+        "x: [*batch, n_columns]",
+        "return: [*batch]",
+    )
+    def batch_mean(x: AnyNDArray) -> AnyNDArray:
+        mean: AnyNDArray = np.mean(x, axis=-1)
+        return mean
+
+    # [dimension_spec_variable_rank__star]
+
+    for batch_shape in [(), (2,), (2, 4)]:
+        x = np.ones(batch_shape + (3,))
+        batch_mean(x)
+
+
+def test_example__dimension_spec_variable_rank__ellipsis() -> None:
+    # [dimension_spec_variable_rank__ellipsis]
+
+    @check_shapes(
+        "x: [batch..., n_columns]",
+        "return: [batch...]",
+    )
+    def batch_mean(x: AnyNDArray) -> AnyNDArray:
+        mean: AnyNDArray = np.mean(x, axis=-1)
+        return mean
+
+    # [dimension_spec_variable_rank__ellipsis]
+
+    for batch_shape in [(), (2,), (2, 4)]:
+        x = np.ones(batch_shape + (3,))
+        batch_mean(x)
+
+
+def test_example__dimension_spec_anonymous_variable_rank__star() -> None:
+    # [dimension_spec_anonymous_variable_rank__star]
+
+    @check_shapes(
+        "x: [*]",
+    )
+    def rank(x: AnyNDArray) -> int:
+        return len(x.shape)
+
+    # [dimension_spec_anonymous_variable_rank__star]
+
+    for batch_shape in [(), (2,), (2, 4)]:
+        x = np.ones(batch_shape)
+        rank(x)
+
+
+def test_example__dimension_spec_anonymous_variable_rank__ellipsis() -> None:
+    # [dimension_spec_anonymous_variable_rank__ellipsis]
+
+    @check_shapes(
+        "x: [...]",
+    )
+    def rank(x: AnyNDArray) -> int:
+        return len(x.shape)
+
+    # [dimension_spec_anonymous_variable_rank__ellipsis]
+
+    for batch_shape in [(), (2,), (2, 4)]:
+        x = np.ones(batch_shape)
+        rank(x)
+
+
+def test_example__dimension_spec__scalar() -> None:
+    # [dimension_spec__scalar]
+
+    @check_shapes(
+        "x: [...]",
+        "return: []",
+    )
+    def mean(x: AnyNDArray) -> AnyNDArray:
+        mean: AnyNDArray = np.sum(x) / x.size
+        return mean
+
+    # [dimension_spec__scalar]
+
+    for batch_shape in [(), (2,), (2, 4)]:
+        x = np.ones(batch_shape)
+        mean(x)
+
+
+def test_example__dimension_spec_broadcast() -> None:
+
+    # [dimension_spec_broadcast]
+
+    @check_shapes(
+        "a: [broadcast batch...]",
+        "b: [broadcast batch...]",
+        "return: [batch...]",
+    )
+    def add(a: AnyNDArray, b: AnyNDArray) -> AnyNDArray:
+        return a + b
+
+    # [dimension_spec_broadcast]
+
+    add(np.ones((3, 1)), np.ones((1, 4)))
+
+
+def test_example__bool_spec_argument_ref() -> None:
+    # pylint: disable=unused-argument
+
+    # [bool_spec_argument_ref]
+
+    @check_shapes(
+        "a: [broadcast batch...] if check_a",
+        "b: [broadcast batch...] if check_b",
+        "return: [batch...]",
+    )
+    def add(a: AnyNDArray, b: AnyNDArray, check_a: bool = True, check_b: bool = True) -> AnyNDArray:
+        return a + b
+
+    # [bool_spec_argument_ref]
+
+    add(np.ones((3, 1)), np.ones((1, 4)), check_b=False)
```

Comment:
I would include the call in the code example to show that the user can set these arguments.
```suggestion

    add(np.ones((3, 1)), np.ones((1, 4)), check_b=False)
    # [bool_spec_argument_ref]
```

Reply:
> I would include the call in the code example to show that the user can set these arguments.

Sure.